clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -556,6 +563,23 @@ public synchronized void subscribePartition(PubSubTopicPartition topicPartition)
     subscribePartition(topicPartition, true);
   }
 
+  // TODO: ensure thread safety for version role change.
+  public synchronized void versionRoleChangeToTriggerResubscribe(TopicPartitionReplicaRole.VersionRole versionRole) {
```

Comment:
a bit confused here, would it be triggered to all backup/current/future SIT to do the re-sub?

Reply:
It would trigger all SIT to do re-sub, when we detect a role change. So you think this is too costly?

Reply:
No, I think I kind of realized they are all necessary at least for now.